crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs

```diff
@@ -20,24 +20,30 @@ pub(crate) struct BuiltInResult {
 }
 
 impl BuiltInResult {
-    pub(super) fn to_ffi_compatible_rust_type(&self, swift_bridge_path: &Path) -> TokenStream {
+    pub(super) fn to_ffi_compatible_rust_type(
+        &self,
+        swift_bridge_path: &Path,
+        types: &TypeDeclarations,
+    ) -> TokenStream {
         if self.is_custom_result_type() {
             let ty = format_ident!("{}", self.c_struct_name());
             return quote! {
                 #ty
             };
         }
+
+        if self.ok_ty.can_be_encoded_with_zero_bytes() {
+            let err_ty = self.err_ty.to_rust_type_path(types);
```

Comment:
I think we want `return self.to_ffi_compatible_rust_type(swift_bridge_path, types);`